<a href="https://colab.research.google.com/github/sithmi4/Statistical-Learning-e23207/blob/main/e23207_data_analysis_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# # **Foundations of Statistical Inference & Hypothesis Testing**

## **Part A: Theoretical Fundamentals (Maximum Likelihood & Decision Space)**

### **1. Bias in the Maximum Likelihood Estimator for Covariance & Bessel's Correction**

#### **Derivation of $\mathbb{E}[\widehat{\boldsymbol{\Sigma}}_{\text{MLE}}] = \frac{n-1}{n}\boldsymbol{\Sigma}$**
Let $\mathbf{X}_1, \mathbf{X}_2, \dots, \mathbf{X}_n \in \mathbb{R}^m$ be i.i.d. random vectors with population mean $\boldsymbol{\mu}$ and population covariance matrix $\boldsymbol{\Sigma} = \mathbb{E}[(\mathbf{X}_i - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T]$.

The empirical sample mean is $\widehat{\boldsymbol{\mu}}_n = \frac{1}{n}\sum_{i=1}^n \mathbf{X}_i$. The raw Maximum Likelihood Estimator (MLE) for the covariance matrix is defined as:
$$\widehat{\boldsymbol{\Sigma}}_{\text{MLE}} = \frac{1}{n} \sum_{i=1}^n (\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)(\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)^T$$

We rewrite the centered sample term using the true population mean $\boldsymbol{\mu}$:
$$\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n = (\mathbf{X}_i - \boldsymbol{\mu}) - (\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})$$

Expanding the outer product inside the summation yields:
$$\sum_{i=1}^n (\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)(\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)^T = \sum_{i=1}^n \left[ (\mathbf{X}_i - \boldsymbol{\mu}) - (\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu}) \right] \left[ (\mathbf{X}_i - \boldsymbol{\mu}) - (\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu}) \right]^T$$
$$= \sum_{i=1}^n (\mathbf{X}_i - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T - \sum_{i=1}^n (\mathbf{X}_i - \boldsymbol{\mu})(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})^T - \sum_{i=1}^n (\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T + \sum_{i=1}^n (\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})^T$$

Since $\sum_{i=1}^n (\mathbf{X}_i - \boldsymbol{\mu}) = n(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})$, the middle cross-product terms simplify:
$$= \sum_{i=1}^n (\mathbf{X}_i - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T - n(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})^T$$

Taking the mathematical expectation on both sides:
$$\mathbb{E}\left[ \sum_{i=1}^n (\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)(\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)^T \right] = \sum_{i=1}^n \mathbb{E}\left[(\mathbf{X}_i - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T\right] - n \mathbb{E}\left[(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})^T\right]$$

By definition:
1. $\mathbb{E}\left[(\mathbf{X}_i - \boldsymbol{\mu})(\mathbf{X}_i - \boldsymbol{\mu})^T\right] = \boldsymbol{\Sigma}$
2. $\mathbb{E}\left[(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu})^T\right] = \text{Var}(\widehat{\boldsymbol{\mu}}_n) = \frac{1}{n}\boldsymbol{\Sigma}$

Substituting these back into our expression:
$$\mathbb{E}\left[ \sum_{i=1}^n (\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)(\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)^T \right] = n\boldsymbol{\Sigma} - n\left(\frac{1}{n}\boldsymbol{\Sigma}\right) = (n-1)\boldsymbol{\Sigma}$$

Thus, dividing by $n$ to find the expected value of $\widehat{\boldsymbol{\Sigma}}_{\text{MLE}}$ proves its downward bias in finite samples:
$$\mathbb{E}[\widehat{\boldsymbol{\Sigma}}_{\text{MLE}}] = \frac{1}{n} \cdot (n-1)\boldsymbol{\Sigma} = \frac{n-1}{n}\boldsymbol{\Sigma}$$

#### **Application of Bessel's Correction**
To eliminate this bias, we multiply the estimator by the scale correction factor $\frac{n}{n-1}$. This adjustment is known as **Bessel's correction**:
$$\mathbf{S} = \frac{n}{n-1}\widehat{\boldsymbol{\Sigma}}_{\text{MLE}} = \frac{1}{n-1}\sum_{i=1}^n (\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)(\mathbf{X}_i - \widehat{\boldsymbol{\mu}}_n)^T$$

Verifying the expectation of $\mathbf{S}$:
$$\mathbb{E}[\mathbf{S}] = \frac{n}{n-1}\mathbb{E}[\widehat{\boldsymbol{\Sigma}}_{\text{MLE}}] = \frac{n}{n-1} \left( \frac{n-1}{n}\boldsymbol{\Sigma} \right) = \boldsymbol{\Sigma}$$
This yields the unique, unbiased sample covariance estimator $\mathbf{S}$.

### **2. Error Metrics and Decision Space Geometric Behavior**

* **Type I Error ($\alpha$):** The probability of declaring a structural anomaly (rejecting $H_0$) when the asset is completely healthy. This represents a **false alarm**.
* **Type II Error ($\beta$):** The probability of failing to flag an anomaly (retaining $H_0$) when the asset is experiencing structural damage ($H_1$). This is a **missed detection**.
* **Consequences of an Ultra-Conservative Threshold ($\alpha = 0.0001$):**
    * **Statistical Power:** Lowering $lpha$ shifts the critical decision boundary further into the distribution's tail. This drastically reduces the probability of rejecting $H_0$ under an alternative state, causing Type II error ($\beta$) to surge and the overall statistical power ($1 - \beta$) of the system to plummet.
    * **Geometric Volume:** The confidence ellipsoid corresponding to "healthy operation" is scaled out aggressively. Because the decision volume scales inversely with risk tolerance, an ultra-conservative $lpha$ expands the boundary, creating a massive ellipsoid volume that absorbs true, low-magnitude structural variations without triggering alerts.

## **Part B: Theoretical Extension (Asymptotic Distributions & Slutsky's Theorem)**

### **1. Rigorous Statement of Slutsky's Theorem**
Let $\mathbf{Y}_n$ and $\mathbf{W}_n$ be sequences of random vectors/matrices. If $\mathbf{Y}_n \xrightarrow{d} \mathbf{Y}$ and $\mathbf{W}_n \xrightarrow{p} \mathbf{C}$ (a constant matrix), then:
1. $\mathbf{Y}_n + \mathbf{W}_n \xrightarrow{d} \mathbf{Y} + \mathbf{C}$
2. \mathbf{W}_n \mathbf{Y}_n \xrightarrow{d} \mathbf{C} \mathbf{Y}
3. \mathbf{W}_n^{-1} \mathbf{Y}_n \xrightarrow{d} \mathbf{C}^{-1} \mathbf{Y} (provided \mathbf{C} is invertible)

### **2. Proof of Operationalized Distribution via Slutsky's**
From the Lindeberg-Lévy Multivariate Central Limit Theorem, we know:
$$\mathbf{Y}_n = \sqrt{n}(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu}) \xrightarrow{d} \mathscr{N}(\mathbf{0}, \boldsymbol{\Sigma})$$

By the Weak Law of Large Numbers (WLLN), the unbiased sample covariance matrix $\mathbf{S}$ is a consistent estimator of the latent population matrix $\boldsymbol{\Sigma}$:
$$\mathbf{S} \xrightarrow{p} \boldsymbol{\Sigma} \implies \mathbf{S}^{-1/2} \xrightarrow{p} \boldsymbol{\Sigma}^{-1/2}$$

Constructing a standardized transformation vector and applying Slutsky's continuous mapping case ($\mathbf{W}_n \mathbf{Y}_n$):
$$\mathbf{S}^{-1/2} \left[ \sqrt{n}(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu}) \right] \xrightarrow{d} \boldsymbol{\Sigma}^{-1/2} \mathscr{N}(\mathbf{0}, \boldsymbol{\Sigma}) \sim \mathscr{N}(\mathbf{0}, \mathbf{I}_{m \times m})$$

For large but finite samples ($n \to \infty$), we reverse this normalization mapping:
$$\sqrt{n}(\widehat{\boldsymbol{\mu}}_n - \boldsymbol{\mu}) \approx \mathbf{S}^{1/2} \mathscr{N}(\mathbf{0}, \mathbf{I}_{m \times m}) \sim \mathscr{N}(\mathbf{0}, \mathbf{S})$$

Dividing by the scalar scaling factor $\sqrt{n}$ centers and shapes the variance landscape, justifying our operational setup:
$$\widehat{\boldsymbol{\mu}}_n \sim \mathscr{N}\left({\boldsymbol{\mu}}, \frac{1}{n}\mathbf{S}\right)$$

## **Part C: Numerical Verification (Python Verification Function)**

In [1]:
import numpy as np
import pandas as pd
import scipy.stats as stats

# --- DO NOT MODIFY: Synthetic Strain Sensor Data Generator ---
np.random.seed(42)
n_samples = 5000
n_features = 3

base_data = np.random.multivariate_normal(
    mean=[0.5, -0.2, 1.1],
    cov=[[0.09, 0.02, 0.01],
         [0.02, 0.06, 0.03],
         [0.01, 0.03, 0.05]],
    size=n_samples
)

# Injecting a subtle, hidden structural drift in the final 1,000 snapshots
base_data[4000:, 0] += 0.015  # Sensor 1 drift
base_data[4000:, 2] -= 0.010  # Sensor 3 drift

df_strain = pd.DataFrame(base_data, columns=['Strain_Ch1', 'Strain_Ch2', 'Strain_Ch3'])
# ----------------------------------------------------------------

def verify_first_moment_homogeneity(df: pd.DataFrame, g_chunks: int = 5) -> dict:
    """
    Partitions the dataset into g chunks and evaluates first-moment homogeneity
    via Wilks' Lambda and Bartlett's Chi-Square asymptotic transformation.
    """
    n = len(df)
    m = df.shape[1]
    chunk_size = n // g_chunks

    # 1. Split df into g equal consecutive blocks
    chunks = [df.iloc[i*chunk_size:(i+1)*chunk_size].to_numpy() for i in range(g_chunks)]

    # 2. Compute global mean and individual chunk means
    global_mean = df.mean().to_numpy()
    chunk_means = [np.mean(chunk, axis=0) for chunk in chunks]

    # 3. Calculate Within-Chunk Matrix (W) and Between-Chunk Matrix (B)
    W = np.zeros((m, m))
    B = np.zeros((m, m))

    for j in range(g_chunks):
        # Within-matrix summation
        diff_W = chunks[j] - chunk_means[j]
        W += diff_W.T @ diff_W

        # Between-matrix summation
        diff_B = chunk_means[j] - global_mean
        B += chunk_size * np.outer(diff_B, diff_B)

    # 4. Compute Wilks' Lambda, Bartlett's Chi-Square, and the final p-value
    det_W = np.linalg.det(W)
    det_Total = np.linalg.det(B + W)
    wilks_lambda = det_W / det_Total

    # Bartlett approximation scaling
    bartlett_factor = n - 1 - (m + g_chunks) / 2.0
    chi2_calc = -bartlett_factor * np.log(wilks_lambda)

    # Degrees of freedom
    dof = m * (g_chunks - 1)
    p_value = 1.0 - stats.chi2.cdf(chi2_calc, dof)

    # Diagnostic interpretation
    alpha = 0.05
    conclusion = (
        "Reject H0: Baseline center has shifted significantly over time (Heterogeneous)."
        if p_value < alpha else
        "Fail to Reject H0: First moment is homogeneous across chunks."
    )

    return {
        "Wilks' Lambda": wilks_lambda,
        "Bartlett Chi-Square": chi2_calc,
        "p-value": p_value,
        "Degrees of Freedom": dof,
        "Conclusion": conclusion
    }

# Execute function
results = verify_first_moment_homogeneity(df_strain, g_chunks=5)
for k, v in results.items():
    print(f"{k}: {v if isinstance(v, str) else f'{v:.6f}'}")

Wilks' Lambda: 0.990452
Bartlett Chi-Square: 47.921505
p-value: 0.000003
Degrees of Freedom: 12.000000
Conclusion: Reject H0: Baseline center has shifted significantly over time (Heterogeneous).


# # **Geometric Subspace Optimization via PCA**

## **Part A: Theoretical Fundamentals (Coordinate Projections & Orthogonality)**

### **1. Covariance Matrix Diagonality Proof**
Given $\mathbf{Z}_i = \mathbf{P}^T \widetilde{\mathbf{X}}_i$ and $\boldsymbol{\Sigma} = \mathbb{E}[\widetilde{\mathbf{X}}_i \widetilde{\mathbf{X}}_i^T] = \mathbf{P} \mathbf{\Lambda} \mathbf{P}^T$, where $\mathbf{P}$ is orthogonal ($\mathbf{P}^T\mathbf{P} = \mathbf{P}\mathbf{P}^T = \mathbf{I}$).

We expand the covariance of the transformed projection matrix:
$$\mathbb{E}[\mathbf{Z}_i \mathbf{Z}_i^T] = \mathbb{E}[(\mathbf{P}^T \widetilde{\mathbf{X}}_i)(\mathbf{P}^T \widetilde{\mathbf{X}}_i)^T] = \mathbb{E}[\mathbf{P}^T \widetilde{\mathbf{X}}_i \widetilde{\mathbf{X}}_i^T \mathbf{P}]$$
Using the linearity of the expectation operator:
$$= \mathbf{P}^T \mathbb{E}[\widetilde{\mathbf{X}}_i \widetilde{\mathbf{X}}_i^T] \mathbf{P} = \mathbf{P}^T \boldsymbol{\Sigma} \mathbf{P}$$
Substituting the spectral decomposition $\boldsymbol{\Sigma} = \mathbf{P} \mathbf{\Lambda} \mathbf{P}^T$:
$$= \mathbf{P}^T (\mathbf{P} \mathbf{\Lambda} \mathbf{P}^T) \mathbf{P} = (\mathbf{P}^T \mathbf{P}) \mathbf{\Lambda} (\mathbf{P}^T \mathbf{P})$$
Since $\mathbf{P}^T\mathbf{P} = \mathbf{I}$:
$$= \mathbf{I} \mathbf{\Lambda} \mathbf{I} = \mathbf{\Lambda}$$

* **Interpretation:** Because $\mathbf{\Lambda}$ is strictly a diagonal matrix, the cross-covariance between any two distinct coordinates $Z_{i,j}$ and $Z_{i,k}$ ($j \neq k$) is exactly zero. Geometrically, PCA performs a rigid rotation of axes that eliminates multi-sensor redundancy and creates mutually orthogonal monitoring directions.

### **2. Conservation of Total Variance via Trace Identities**
Using the trace cyclic permutation property ($\text{tr}(\mathbf{A}\mathbf{B}\mathbf{C}) = \text{tr}(\mathbf{C}\mathbf{A}\mathbf{B})$):
$$\text{tr}(\boldsymbol{\Sigma}) = \text{tr}(\mathbf{P} \mathbf{\Lambda} \mathbf{P}^T) = \text{tr}(\mathbf{P}^T \mathbf{P} \mathbf{\Lambda}) = \text{tr}(\mathbf{I} \mathbf{\Lambda}) = \text{tr}(\mathbf{\Lambda}) = \sum_{j=1}^m \lambda_j$$
This shows that transforming coordinates to principal components reallocates variance without gaining or losing information.

* **Subspace Ratios:**
    * **Cumulative Explained Variance Ratio:** $\Phi(k) = \frac{\sum_{j=1}^k \widehat{\lambda}_j}{\sum_{j=1}^m \widehat{\lambda}_j}$
    * **Residual Unexplained Variance Ratio:** $\Psi(k) = 1 - \Phi(k) = \frac{\sum_{j=k+1}^m \widehat{\lambda}_j}{\sum_{j=1}^m \widehat{\lambda}_j}$

### **3. Subspace Energy Decomposition Proof**
Let $(\mathbf{x}_i - \widehat{\boldsymbol{\mu}}_n) = \widehat{\mathbf{P}}_k \mathbf{z}_{i,k} + \widehat{\mathbf{P}}_{m-k} \mathbf{z}_{i,m-k} = \widehat{\mathbf{x}}_i + \mathbf{e}_i$.
$$\|\mathbf{x}_i - \widehat{\boldsymbol{\mu}}_n\|^2 = (\widehat{\mathbf{P}}_k \mathbf{z}_{i,k} + \widehat{\mathbf{P}}_{m-k} \mathbf{z}_{i,m-k})^T (\widehat{\mathbf{P}}_k \mathbf{z}_{i,k} + \widehat{\mathbf{P}}_{m-k} \mathbf{z}_{i,m-k})$$
$$= \mathbf{z}_{i,k}^T \widehat{\mathbf{P}}_k^T \widehat{\mathbf{P}}_k \mathbf{z}_{i,k} + 2\mathbf{z}_{i,k}^T \widehat{\mathbf{P}}_k^T \widehat{\mathbf{P}}_{m-k} \mathbf{z}_{i,m-k} + \mathbf{z}_{i,m-k}^T \widehat{\mathbf{P}}_{m-k}^T \widehat{\mathbf{P}}_{m-k} \mathbf{z}_{i,m-k}$$
Because the eigenvectors are mutually orthogonal, $\widehat{\mathbf{P}}_k^T \widehat{\mathbf{P}}_k = \mathbf{I}_k$, $\widehat{\mathbf{P}}_{m-k}^T \widehat{\mathbf{P}}_{m-k} = \mathbf{I}_{m-k}$, and the cross-subspace term $\widehat{\mathbf{P}}_k^T \widehat{\mathbf{P}}_{m-k} = \mathbf{0}$. The expression simplifies to independent subspace energies:
$$= \mathbf{z}_{i,k}^T \mathbf{z}_{i,k} + \mathbf{z}_{i,m-k}^T \mathbf{z}_{i,m-k} = \|\mathbf{z}_{i,k}\|^2 + \|\mathbf{z}_{i,m-k}\|^2$$

* **Diagnostic Sensitivity Matching:**
    * **Environmental Load Anomaly $\to$ Hotelling's $T^2$:** Extreme wind or operational loads increase operational amplitudes but follow standard system dynamics. This forces larger projections onto existing primary axes, resulting in a spike in $T^2$.
    * **Internal Structural Fracture $	o$ Residual $Q$ Statistic:** Structural cracks change the underlying physical load pathways, modifying the covariance structure of the sensors. The signals no longer align with the trained PCA subspace, causing energy to spill into the residual space and triggering a major spike in the $Q$ statistic.

## **Part B: Numerical Verification (Python Class Method Implementation)**

In [4]:
import numpy as np

class PCADiagnosticsEngine:
    def __init__(self, data: np.ndarray):
        self.raw_data = data
        self.n, self.m = data.shape

    def execute_pipeline(self) -> dict:
        # 1. Standardize and center the incoming realizations (ddof=1)
        mean = np.mean(self.raw_data, axis=0)
        std = np.std(self.raw_data, axis=0, ddof=1)
        std = np.where(std == 0, 1e-15, std)
        Z = (self.raw_data - mean) / std

        # 2. Decompose unbiased sample covariance matrix S
        S = np.cov(Z, rowvar=False)
        eigenvalues, eigenvectors = np.linalg.eigh(S)

        # Sort in descending order
        idx = np.argsort(eigenvalues)[::-1]
        evals = eigenvalues[idx]
        evecs = eigenvectors[:, idx]

        # Project data into score coordinates
        scores = Z @ evecs

        results = {}
        # 3. Loop through every possible subspace cutoff value k
        for k in range(1, self.m):
            # Hotelling's T^2 calculation per snapshot
            t2_i = np.sum((scores[:, :k] ** 2) / evals[:k], axis=1)

            # Q statistic (SPE) calculation per snapshot
            q_i = np.sum(scores[:, k:] ** 2, axis=1)

            # 4. Compute expected values across all samples
            results[f"k={k}"] = {
                "Mean_T2": np.mean(t2_i),
                "Mean_Q": np.mean(q_i)
            }

        return results

# Verification script with mock asset archive data
np.random.seed(42)
mock_archive = np.random.normal(0, 1, (3000, 4))
engine = PCADiagnosticsEngine(mock_archive)
metrics = engine.execute_pipeline()
for k, vals in metrics.items():
    print(f"Subspace Cutoff {k} -> Expected T2: {vals['Mean_T2']:.4f} | Expected Q: {vals['Mean_Q']:.4f}")

Subspace Cutoff k=1 -> Expected T2: 0.9997 | Expected Q: 2.9744
Subspace Cutoff k=2 -> Expected T2: 1.9993 | Expected Q: 1.9617
Subspace Cutoff k=3 -> Expected T2: 2.9990 | Expected Q: 0.9679


# # **Latent Subspace Decomposition via Factor Analysis (FA)**

## **Part A: Theoretical Fundamentals (Generative Model & Commonalities)**

### **1. Fundamental Equation of Factor Analysis Proof**
Given the generative model $\mathbf{Z}_i = \boldsymbol{\Lambda} \mathbf{F}_i + \mathbf{\epsilon}_i$. We expand the matrix correlation structure $\mathbf{R}$:
$$\mathbf{R} = \mathbb{E}[\mathbf{Z}_i \mathbf{Z}_i^T] = \mathbb{E}[(\boldsymbol{\Lambda} \mathbf{F}_i + \mathbf{\epsilon}_i)(\boldsymbol{\Lambda} \mathbf{F}_i + \mathbf{\epsilon}_i)^T]$$
$$= \mathbb{E}[\boldsymbol{\Lambda} \mathbf{F}_i \mathbf{F}_i^T \boldsymbol{\Lambda}^T + \boldsymbol{\Lambda} \mathbf{F}_i \mathbf{\epsilon}_i^T + \mathbf{\epsilon}_i \mathbf{F}_i^T \boldsymbol{\Lambda}^T + \mathbf{\epsilon}_i \mathbf{\epsilon}_i^T]$$
Using the linearity of expectation and pulling out deterministic matrices:
$$= \boldsymbol{\Lambda} \mathbb{E}[\mathbf{F}_i \mathbf{F}_i^T] \boldsymbol{\Lambda}^T + \boldsymbol{\Lambda} \mathbb{E}[\mathbf{F}_i \mathbf{\epsilon}_i^T] + \mathbb{E}[\mathbf{\epsilon}_i \mathbf{F}_i^T] \boldsymbol{\Lambda}^T + \mathbb{E}[\mathbf{\epsilon}_i \mathbf{\epsilon}_i^T]$$
Applying the model rules ($\mathbb{E}[\mathbf{F}_i \mathbf{F}_i^T] = \mathbf{I}$, $\mathbb{E}[\mathbf{\epsilon}_i \mathbf{\epsilon}_i^T] = \boldsymbol{\Psi}$, and factor-to-noise independence $\mathbb{E}[\mathbf{\epsilon}_i \mathbf{F}_i^T] = \mathbf{0}$):
$$\mathbf{R} = \boldsymbol{\Lambda} (\mathbf{I}) \boldsymbol{\Lambda}^T + \mathbf{0} + \mathbf{0} + \boldsymbol{\Psi} = \boldsymbol{\Lambda} \boldsymbol{\Lambda}^T + \boldsymbol{\Psi}$$

* **Physical Meanings:**
    * **Communality ($h_j^2 = \sum_{r=1}^k \lambda_{j,r}^2$):** The portion of a sensor's variance explained by global physical processes (shared structural loads). High communality means the sensor reliably tracks global asset health.
    * **Uniqueness ($\varphi_j^2 = \Psi_{jj}$):** The isolated, localized variance of a sensor (such as high-frequency noise or local thermal variation). If uniqueness climbs toward $100\%$, it points to an instrumentation fault rather than systemic structural damage.

### **2. Coordinate Rotation Dynamics**
* **PCA vs. Varimax-Rotated FA:** PCA eigenvectors maximize variance across a strict hierarchy, often leading to complex loadings where every sensor maps partially to every component. Varimax rotations maximize the variance of the *squared* loadings within each column. This drives loadings toward either 1 or 0, grouping sensors onto clear, individual factors.
* **Invariance Under Orthogonal Rotations:** Let $\boldsymbol{\Lambda}^* = \boldsymbol{\Lambda}\mathbf{T}$ be the rotated loading matrix, where $\mathbf{T}$ is an orthogonal transformation matrix ($\mathbf{T}\mathbf{T}^T = \mathbf{I}$). Expanding the rotated covariance approximation yields:
$$\boldsymbol{\Lambda}^* (\boldsymbol{\Lambda}^*)^T + \boldsymbol{\Psi} = (\boldsymbol{\Lambda}\mathbf{T})(\boldsymbol{\Lambda}\mathbf{T})^T + \boldsymbol{\Psi} = \boldsymbol{\Lambda}(\mathbf{T}\mathbf{T}^T)\boldsymbol{\Lambda}^T + \boldsymbol{\Psi} = \boldsymbol{\Lambda}\mathbf{I}\boldsymbol{\Lambda}^T + \boldsymbol{\Psi} = \boldsymbol{\Lambda}\boldsymbol{\Lambda}^T + \boldsymbol{\Psi} \approx \mathbf{R}$$
Because $\mathbf{T}\mathbf{T}^T = \mathbf{I}$, rotating the space reallocates loadings across columns without changing individual sensor communalities ($h_j^2$) or the total covariance fit.

### **3. Thomson's Regression Method Proof**
#### **Joint Distribution Expansion**
We build the joint vector $\mathbf{Y}_i = [\mathbf{Z}_i^T, \mathbf{F}_i^T]^T$. The joint covariance matrix blocks are defined as:
$$\boldsymbol{\Sigma}_{11} = \mathbb{E}[\mathbf{Z}_i \mathbf{Z}_i^T] = \mathbf{R}, \quad \boldsymbol{\Sigma}_{22} = \mathbb{E}[\mathbf{F}_i \mathbf{F}_i^T] = \mathbf{I}_{k \times k}$$
$$\boldsymbol{\Sigma}_{21} = \mathbb{E}[\mathbf{F}_i \mathbf{Z}_i^T] = \mathbb{E}[\mathbf{F}_i (\boldsymbol{\Lambda}\mathbf{F}_i + \mathbf{\epsilon}_i)^T] = \mathbb{E}[\mathbf{F}_i \mathbf{F}_i^T \boldsymbol{\Lambda}^T + \mathbf{F}_i \mathbf{\epsilon}_i^T] = \mathbf{I}\boldsymbol{\Lambda}^T + \mathbf{0} = \boldsymbol{\Lambda}^T$$
$$\boldsymbol{\Sigma}_{12} = \boldsymbol{\Sigma}_{21}^T = \boldsymbol{\Lambda}$$
Assembling these blocks yields the joint covariance structure:
$$\boldsymbol{\Sigma}_{YY} = \begin{bmatrix} \mathbf{R} & \boldsymbol{\Lambda} \\ \boldsymbol{\Lambda}^T & \mathbf{I}_{k \times k} \end{bmatrix}$$

#### **Conditional Expectation Derivation**
Using the Gaussian conditional projection rule, since variables are centered ($\boldsymbol{\mu}_Z = \mathbf{0}, \boldsymbol{\mu}_F = \mathbf{0}$):
$$\mathbf{f}_i = \mathbb{E}[\mathbf{F}_i | \mathbf{z}_i] = \mathbf{0} + \boldsymbol{\Sigma}_{21} \boldsymbol{\Sigma}_{11}^{-1} (\mathbf{z}_i - \mathbf{0})$$
Substituting the joint matrix blocks $\boldsymbol{\Sigma}_{21} = \boldsymbol{\Lambda}^T$ and \boldsymbol{\Sigma}_{11}^{-1} = \mathbf{R}^{-1} completes Thomson's factor score derivation:
$$\mathbf{f}_i = \boldsymbol{\Lambda}^T \mathbf{R}^{-1} \mathbf{z}_i$$

## **Part B: Parametric Factor Partition Engine & UI Dashboard Pipeline**

In [3]:
import numpy as np
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from sklearn.decomposition import FactorAnalysis

# --- RUN DATA GENERATION INTERFACE ---
np.random.seed(42)
n_samples = 2500
f1 = np.random.normal(0, 1, n_samples)
f2 = np.random.normal(0, 1, n_samples)

s1 = 0.85 * f1 + 0.10 * f2 + np.random.normal(0, 0.3, n_samples)
s2 = 0.80 * f1 + 0.15 * f2 + np.random.normal(0, 0.35, n_samples)
s3 = 0.12 * f1 + 0.90 * f2 + np.random.normal(0, 0.25, n_samples)
s4 = 0.02 * f1 + 0.05 * f2 + np.random.normal(0, 1.40, n_samples)

df_asset = pd.DataFrame(
    data=np.vstack([s1, s2, s3, s4]).T,
    columns=['Sensor_1', 'Sensor_2', 'Sensor_3', 'Sensor_4']
)

def run_factor_diagnostics_pipeline(df: pd.DataFrame, k_factors: int = 2):
    n, m = df.shape

    # 1. Z-Score Mapping Layer with Zero-Variance Guardrails
    stds = df.std(ddof=1).to_numpy()
    stds = np.where(stds == 0, 1e-15, stds)
    Z = (df.to_numpy() - df.mean().to_numpy()) / stds

    # 2. Dimensionality Constraint Layer Validation Checks
    if n <= m:
        raise ValueError("Rule Exception: Snapshot count (n) must exceed channel count (m).")
    if k_factors >= m:
        raise ValueError("Rule Exception: Latent spaces (k) must be strictly less than feature dimension (m).")

    # 3. Information Decomposition Layer using Varimax-Rotated FA
    fa = FactorAnalysis(n_components=k_factors, rotation='varimax', random_state=42)
    fa.fit(Z)

    lambda_rotated = fa.components_.T # Loadings matrix (m x k)
    uniqueness = fa.noise_variance_   # Unique variance vector (phi^2)
    communality = 1.0 - uniqueness     # Communality vector (h^2)

    # Latent Score Projection via Thomson's Method
    R_empirical = np.corrcoef(Z, rowvar=False)
    R_inv = np.linalg.inv(R_empirical)
    factor_scores = Z @ R_inv @ lambda_rotated
    factor_variances = np.var(factor_scores, axis=0, ddof=1)

    # Print Performance Diagnostic Telemetry to Logging Console
    avg_comm = np.mean(communality) * 100
    avg_uniq = np.mean(uniqueness) * 100
    print(f"--- TELEMETRY METRICS ---")
    print(f"Average System Communality %: {avg_comm:.2f}%")
    print(f"Average System Uniqueness %: {avg_uniq:.2f}%\n")

    # --- INTERFACE DESIGN: 2 x 2 SUBPLOT CANVAS ---
    fig = make_subplots(
        rows=2, cols=2,
        horizontal_spacing=0.24,
        vertical_spacing=0.28,
        subplot_titles=(
            "Structural Loadings Matrix Heatmap",
            "Variance Partitioning Profile",
            "Sensor Uniqueness Line Profile",
            "Latent Factor Variance Distribution"
        )
    )

    # Subplot (1,1): Structural Loadings Matrix Heatmap
    fig.add_trace(
        go.Heatmap(
            z=np.abs(lambda_rotated),
            x=[f"Factor_{j+1}" for j in range(k_factors)],
            y=df.columns.tolist(),
            colorscale='YlOrRd',
            colorbar=dict(
                x=-0.15, len=0.38, y=0.78,
                yanchor="middle", xanchor="right"
            ),
            showscale=True
        ), row=1, col=1
    )

    # Subplot (1,2): Variance Partitioning Profile (Horizontal Stacked Bar)
    fig.add_trace(
        go.Bar(
            y=df.columns.tolist(),
            x=communality * 100,
            orientation='h',
            name='Shared Communality (h²)',
            marker=dict(color='#1f77b4')
        ), row=1, col=2
    )
    fig.add_trace(
        go.Bar(
            y=df.columns.tolist(),
            x=uniqueness * 100,
            orientation='h',
            name='Isolated Uniqueness (φ²)',
            marker=dict(color='#ff7f0e')
        ), row=1, col=2
    )

    # Subplot (2,1): Sensor Uniqueness Line Profile
    fig.add_trace(
        go.Scatter(
            x=df.columns.tolist(),
            y=uniqueness,
            mode='lines+markers',
            name='Uniqueness (φ²)',
            line=dict(color='#d62728', width=2, dash='dashdot'),
            marker=dict(symbol='x', size=8)
        ), row=2, col=1
    )

    # Subplot (2,2): Latent Factor Variance Distribution
    fig.add_trace(
        go.Bar(
            x=[f"Factor_{j+1}" for j in range(k_factors)],
            y=factor_variances,
            name='Factor Score Variance',
            marker=dict(color='#2ca02c', line=dict(color='black', width=0.5))
        ), row=2, col=2
    )

    # Global Canvas Configurations
    fig.update_layout(
        width=1250, height=750,
        template='plotly_white',
        barmode='stack',
        showlegend=True,
        legend=dict(orientation="h", x=0.5, y=1.12, xanchor="center"),
        margin=dict(t=150, b=60, l=140, r=80)
    )
    fig.update_xaxes(range=[0, 100], row=1, col=2)
    fig.update_xaxes(tickangle=25, row=2, col=1)

    fig.show()

# Execute diagnostics dashboard layout
run_factor_diagnostics_pipeline(df_asset, k_factors=2)

--- TELEMETRY METRICS ---
Average System Communality %: 50.65%
Average System Uniqueness %: 49.35%



---
---

# Assignment 4 — Subspace Diagnostics & Feature De-correlation

---

## Part A: Feature Analysis Dashboard (PCA 2×3 + FA 2×2)


In [5]:
import numpy as np
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from sklearn.decomposition import FactorAnalysis
import warnings
warnings.filterwarnings('ignore')

# ──────────────────────────────────────────────────────────────
# SYNTHETIC DATA GENERATOR (do not modify)
# ──────────────────────────────────────────────────────────────
np.random.seed(42)
n_samples = 2500

f1 = np.random.normal(0, 1, n_samples)
f2 = np.random.normal(0, 1, n_samples)

s1 = 0.85 * f1 + 0.10 * f2 + np.random.normal(0, 0.30, n_samples)
s2 = 0.80 * f1 + 0.15 * f2 + np.random.normal(0, 0.35, n_samples)
s3 = 0.12 * f1 + 0.90 * f2 + np.random.normal(0, 0.25, n_samples)
s4 = 0.02 * f1 + 0.05 * f2 + np.random.normal(0, 1.40, n_samples)

df_asset = pd.DataFrame(
    data=np.vstack([s1, s2, s3, s4]).T,
    columns=['Sensor_1', 'Sensor_2', 'Sensor_3', 'Sensor_4']
)

# ──────────────────────────────────────────────────────────────
# HELPER — PCA ENGINE
# ──────────────────────────────────────────────────────────────
def run_pca(df):
    X  = df.values
    mu = X.mean(axis=0)
    Xc = X - mu
    S  = np.cov(Xc, rowvar=False, ddof=1)
    evals, evecs = np.linalg.eigh(S)
    idx   = np.argsort(evals)[::-1]
    evals = evals[idx];  evecs = evecs[:, idx]
    Z     = Xc @ evecs   # full score matrix

    m    = len(evals)
    k_range = list(range(1, m))   # [1, 2, 3] for m=4

    T2_means, Q_means, expl_ratios = [], [], []
    for k in k_range:
        Zk   = Z[:, :k]
        Zres = Z[:, k:]
        T2 = np.sum(Zk**2 / evals[:k], axis=1)
        Q  = np.sum(Zres**2, axis=1)
        T2_means.append(T2.mean())
        Q_means.append(Q.mean())
        expl_ratios.append(evals[:k].sum() / evals.sum())

    return {
        'evals'      : evals,
        'evecs'      : evecs,
        'expl_ratios': np.array(expl_ratios),
        'k_range'    : k_range,
        'T2_means'   : np.array(T2_means),
        'Q_means'    : np.array(Q_means),
    }

# ──────────────────────────────────────────────────────────────
# HELPER — FA ENGINE
# ──────────────────────────────────────────────────────────────
def run_fa(df, n_factors=2):
    numeric = df.select_dtypes(include=np.number)
    mu  = numeric.mean(axis=0).values
    std = numeric.std(axis=0, ddof=1).values
    std = np.where(std == 0, 1e-15, std)
    Z   = (numeric.values - mu) / std
    n, m = Z.shape

    fa = FactorAnalysis(n_components=n_factors, rotation='varimax',
                        svd_method='lapack', random_state=42)
    fa.fit(Z)

    loadings    = fa.components_.T   # (m, k)
    uniqueness  = fa.noise_variance_
    communality = 1.0 - uniqueness

    R     = np.corrcoef(Z, rowvar=False)
    R_inv = np.linalg.inv(R)
    scores = Z @ R_inv @ loadings
    factor_var = scores.var(axis=0, ddof=1)

    return {
        'loadings'   : loadings,
        'communality': communality,
        'uniqueness' : uniqueness,
        'factor_var' : factor_var,
        'sensors'    : list(numeric.columns),
    }

# ──────────────────────────────────────────────────────────────
# RUN BOTH ENGINES
# ──────────────────────────────────────────────────────────────
pca = run_pca(df_asset)
fa  = run_fa(df_asset, n_factors=2)

sensors = [f'Sensor_{i+1}' for i in range(4)]
factors = ['Factor 1', 'Factor 2']

# ══════════════════════════════════════════════════════════════
# DASHBOARD 1: PCA 2×3
# ══════════════════════════════════════════════════════════════
fig_pca = make_subplots(
    rows=2, cols=3,
    subplot_titles=[
        'Feature Loadings |P_jr|',
        'Absolute Eigenvalues',
        'Marginal & Cumulative Explained Variance %',
        'Residual Space (Unexplained %)',
        'Mean Hotelling T² vs k',
        'Mean Q Statistic (SPE) vs k',
    ],
    horizontal_spacing=0.12,
    vertical_spacing=0.26,
)

m   = 4
k_r = pca['k_range']       # [1, 2, 3]
k_labels = [f'k={k}' for k in k_r]

# (1,1) Loadings heatmap
L = np.abs(pca['evecs'])    # (m, m)
pc_labels = [f'PC{j+1}' for j in range(m)]
fig_pca.add_trace(
    go.Heatmap(z=L, x=pc_labels, y=sensors, colorscale='YlOrRd', showscale=True,
               text=np.round(L,3), texttemplate='%{text}',
               colorbar=dict(x=0.28, len=0.45, y=0.78)),
    row=1, col=1)

# (1,2) Eigenvalues bar
fig_pca.add_trace(
    go.Bar(x=pc_labels, y=pca['evals'], marker_color='steelblue',
           name='Eigenvalue', showlegend=False),
    row=1, col=2)

# (1,3) Marginal % bars + cumulative line
marginal_pct = np.diff(np.concatenate([[0], pca['expl_ratios']])) * 100
cum_pct      = pca['expl_ratios'] * 100
fig_pca.add_trace(
    go.Bar(x=k_labels, y=marginal_pct, name='Marginal Var %',
           marker_color='#aec7e8', showlegend=True),
    row=1, col=3)
fig_pca.add_trace(
    go.Scatter(x=k_labels, y=cum_pct, mode='lines+markers',
               line=dict(dash='dash', color='navy'),
               name='Cumulative Var %', showlegend=True),
    row=1, col=3)

# (2,1) Residual %
residual_pct = 100 - cum_pct
fig_pca.add_trace(
    go.Bar(x=k_labels, y=residual_pct, marker_color='#d62728',
           name='Unexplained %', showlegend=False),
    row=2, col=1)

# (2,2) Mean T²
fig_pca.add_trace(
    go.Scatter(x=k_labels, y=pca['T2_means'], mode='lines+markers',
               marker=dict(size=8), line=dict(color='darkorange'),
               name='Mean T²', showlegend=False),
    row=2, col=2)

# (2,3) Mean Q
fig_pca.add_trace(
    go.Scatter(x=k_labels, y=pca['Q_means'], mode='lines+markers',
               marker=dict(size=8), line=dict(color='green'),
               name='Mean Q (SPE)', showlegend=False),
    row=2, col=3)

fig_pca.update_layout(
    width=1350, height=780,
    template='plotly_white',
    title_text='PCA Optimization Suite — 2×3 Dashboard',
    title_x=0.5,
    legend=dict(orientation='h', x=0.5, y=1.05, xanchor='center'),
    margin=dict(t=140, b=60, l=80, r=60),
)
fig_pca.show()

# ══════════════════════════════════════════════════════════════
# DASHBOARD 2: FA 2×2
# ══════════════════════════════════════════════════════════════
fig_fa = make_subplots(
    rows=2, cols=2,
    subplot_titles=[
        'Structural Loadings Heatmap |λ|',
        'Variance Partitioning (h² vs φ²)',
        'Sensor Uniqueness Noise Floor (φ²)',
        'Latent Factor Scores Empirical Variance',
    ],
    horizontal_spacing=0.24,
    vertical_spacing=0.28,
)

L_fa = np.abs(fa['loadings'])    # (m=4, k=2)
comm_pct = fa['communality'] * 100
uniq_pct = fa['uniqueness']  * 100

# (1,1) Loadings heatmap
fig_fa.add_trace(
    go.Heatmap(z=L_fa, x=factors, y=sensors, colorscale='YlOrRd', showscale=True,
               text=np.round(L_fa,3), texttemplate='%{text}',
               colorbar=dict(x=-0.15, len=0.38, y=0.78,
                             yanchor='middle', xanchor='right')),
    row=1, col=1)

# (1,2) Stacked horizontal bar
fig_fa.add_trace(
    go.Bar(y=sensors, x=comm_pct, orientation='h',
           name='Shared Communality (h²)', marker_color='#1f77b4', showlegend=True),
    row=1, col=2)
fig_fa.add_trace(
    go.Bar(y=sensors, x=uniq_pct, orientation='h',
           name='Isolated Uniqueness (φ²)', marker_color='#ff7f0e', showlegend=True),
    row=1, col=2)
fig_fa.update_layout(barmode='stack')
fig_fa.update_xaxes(range=[0, 100], row=1, col=2)

# (2,1) Uniqueness line
fig_fa.add_trace(
    go.Scatter(x=sensors, y=fa['uniqueness'], mode='lines+markers',
               line=dict(color='#d62728', width=2, dash='dashdot'),
               marker=dict(symbol='x', size=10),
               name='φ²', showlegend=False),
    row=2, col=1)
fig_fa.update_xaxes(tickangle=25, row=2, col=1)

# (2,2) Factor variance bars
fig_fa.add_trace(
    go.Bar(x=factors, y=fa['factor_var'],
           marker_color='#2ca02c',
           marker_line=dict(color='black', width=0.5),
           name='Factor Variance', showlegend=False),
    row=2, col=2)

fig_fa.update_layout(
    width=1250, height=750,
    template='plotly_white',
    title_text='FA Latent Subspace Suite — 2×2 Dashboard',
    title_x=0.5,
    legend=dict(orientation='h', x=0.5, y=1.02, xanchor='center', yanchor='bottom'),
    margin=dict(t=150, b=60, l=140, r=80),
)
fig_fa.show()

# Print metrics for reference
print("\n=== PCA Results ===")
print(f"Eigenvalues: {np.round(pca['evals'],4)}")
print(f"Explained Var Ratios: {np.round(pca['expl_ratios']*100,2)}%")
print(f"Mean T²:  {np.round(pca['T2_means'],4)}")
print(f"Mean Q:   {np.round(pca['Q_means'],4)}")

print("\n=== FA Results ===")
for s, h, phi in zip(sensors, fa['communality'], fa['uniqueness']):
    print(f"  {s}: h²={h:.4f} ({h*100:.1f}%), φ²={phi:.4f} ({phi*100:.1f}%)")
print(f"Factor Score Variances: {np.round(fa['factor_var'],4)}")



=== PCA Results ===
Eigenvalues: [2.036  1.5424 0.7819 0.1071]
Explained Var Ratios: [45.57 80.1  97.6 ]%
Mean T²:  [0.9996 1.9992 2.9988]
Mean Q:   [2.4304 0.8886 0.1071]

=== FA Results ===
  Sensor_1: h²=0.8650 (86.5%), φ²=0.1350 (13.5%)
  Sensor_2: h²=0.8756 (87.6%), φ²=0.1244 (12.4%)
  Sensor_3: h²=0.2764 (27.6%), φ²=0.7236 (72.4%)
  Sensor_4: h²=0.0091 (0.9%), φ²=0.9909 (99.1%)
Factor Score Variances: [0.925  0.2841]


# # **Subspace Diagnostics & Feature De-correlation (Analysis Answers)**

### **Question 1: The Total Variance Illusion**
1. **Physical Nature of Sensor 4:** A uniqueness value above $98\%$ signals that Sensor 4 shares almost zero variance with the rest of the asset. It operates independently of the shared physical processes ($f_1, f_2$). In practice, this means the sensor's output is almost purely localized instrumentation noise or high-frequency interference.
2. **How PCA is Fooled:** PCA searches for directions that maximize *total global variance*, without separating systemic structural patterns from localized noise. Because Sensor 4 has an exceptionally large variance ($\sigma^2 \approx 2.0$), PCA aligns a primary principal component directly with Sensor 4's noise axis to maximize global variance. This inflates the top eigenvalues, mistaking raw noise for structural data.
3. **Risk to Automated Systems:** If an engineer relies solely on PCA, an instrumentation fault like a short circuit or calibration loss will spike Hotelling's $T^2$ or skew the baseline coordinates. This can lead to costly false alarms that shut down healthy operations, or blind the system to genuine structural failures by masking them with sensor noise.

### **Question 2: Decoupling Structural Loading via Rotation**
1. **Varimax Simple Structure:** Traditional PCA relies on a strict mathematical hierarchy where the first component captures the largest possible variance across all channels, often leading to complex, mixed loadings. Varimax rotation relaxes this hierarchy by rotating the orthogonal axes to maximize the variance of the *squared* loadings within each column. This drives the loadings toward 1 or 0, grouping channels into distinct blocks.
2. **Operator Interpretation:** The rotated FA heatmap is much easier to interpret during an emergency. Instead of dealing with mixed PCA eigenvector coefficients, an operator can instantly isolate the failure: a spike on Factor 1 points directly to a problem with the primary structural system (Sensors 1 and 2), while a spike on Factor 2 isolates the secondary operational mode (Sensor 3).

### **Question 3: Determining Subspace Truncation ($k$)**
1. **Behavior of Mean $Q$ Curve:** When moving from $k=1$ to $k=2$, the Mean $Q$ statistic drops sharply as the second major physical process ($f_2$) is captured. Transitioning from $k=2$ to $k=3$ produces a flat, minor change, forming a distinct "elbow."
2. **Identifying True Dimensionality ($k=2$):** The sharp drop shows that adding the second component captures meaningful structural information. The subsequent flat elbow indicates that adding more components yields diminishing returns, only absorbing minor, unstructured background noise. This identifies $k=2$ as the true dimensionality of the underlying physical system.
3. **Risk of Over-Truncation ($k=3$):** Choosing $k=3$ forces the pure, localized instrumentation noise of Sensor 4 into your "clean" monitoring subspace. This corrupts your structural tracking layer with sensor noise, reducing its sensitivity to actual asset degradation.

### **Question 4: Operational Trade-offs in System Health Monitoring**
* **The FA Strategy is More Robust:** The Factor Analysis strategy provides better protection against individual sensor failures or electrical shorts.
* **Justification:** PCA mixes all sensor variances together; a massive noise spike on Sensor 4 distorts the entire coordinate mapping, throwing off both $T^2$ and $Q$ alerts. Factor Analysis handles this by isolating unique localized variances in its diagonal noise matrix ($\boldsymbol{\Psi}$). Because the engine explicitly assigns Sensor 4 a high uniqueness value ($\varphi_4^2 > 98\%$), it isolates this instrument noise from the shared physical factors. This keeps the structural health tracking layer clean and prevents sensor faults from triggering false plant shutdowns.